### DISCUSSION TITLE
Grunt, Gulp, Webpack, Other?

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#15] [General]
Frontend q, which is the go to option these days?

**Comment by nathanwoulfe** [comment]
Really depends on what you're building. Grunt and Gulp will work fine, but are a bit dated and may start to see issues with support for plugins. Webpack can be a crazy complicated beast, with a stack more features than probably needed for a 'normal' website. 

My last build I used Parcel.js. Virtually no config, just stays out of the way and does what it needs to do. Would recommend.

**Reply by TimGeyssens** [reply]
Thanks!

**Comment by jrcpereira** [comment]
Why not plain Npm scripts and a couple of packages for what you need?

Unless you have more complicated tasks than minifying, bundling and copying tasks, I believe it's the better (simpler) option.

**Reply by TimGeyssens** [reply]
I'm a noob in that area... but yes, simple is better!

**Reply by CarlSargunar** [reply]
+1 for lowest barrier to entry. For me this is a learning tool as much as a best practice tool, but we have to keep barriers to entry as low as poss

**Comment by markadrake** [comment]
Webpack can also be “no config”. So if it comes down to Webpack or Parcel I’d recommend building it in a way that either works. I’d love to help your team explore that as an option. It’s 5am so I might be dreaming :)

Another thought to consider is who this is made for and will be using it. If you want high adoption you may not want to force anyone into this at all.

And with the inevitable release of Bootstrap 5 it might be good to consider it as your FED framework; which you could mimic whatever their build process is so you’re not adding anything unique to this project that doesn’t come from bootstrap 5.

**Reply by TimGeyssens** [reply]
hey @markadrake we have a #8 goal discussion going :) , high adoption isn't the plan, just a good learning tool and possible vs template

**Comment by louisjrdev** [comment]
I think something that could easily be swapped in / out, maybe parcel and webpack?
I have used both and parcel is a lot simpler.

**Reply by markadrake** [reply]
Ditto on the being able to swap the 2 in and out - if there is a way we can do 0-config webpack and 0-config parcel and there remain an option to use one or the other with nominal work that would be great! 

For reference: [webpack](https://webpack.js.org/configuration/) & [parcel](https://parceljs.org/getting_started.html) intros.

**Comment by filipbech** [comment]
A good place to learn more and compare the tools, is this project https://tooling.report by some of the chrome people...

**Reply by TimGeyssens** [reply]
cool, thanks for the tip!
